crates/aptos-dkg/src/pvss/fiat_shamir.rs

```diff
@@ -7,52 +7,119 @@
 //! I suspect the reason they want the developer to do things these ways is to force them to cleanly
 //! define all the things that are appended to the transcript.
 
+use crate::range_proof;
 use crate::{
     pvss::{threshold_config::ThresholdConfig, traits::Transcript},
-    utils::random::random_scalar_from_uniform_bytes,
-    SCALAR_NUM_BYTES,
+    // utils::random::random_scalar_from_uniform_bytes,
 };
 use aptos_crypto::ValidCryptoMaterial;
-use blstrs::Scalar;
-use ff::PrimeField;
+use blstrs::G1Projective;
 use serde::Serialize;
 
-pub const PVSS_DOM_SEP: &[u8; 21] = b"APTOS_SCRAPE_PVSS_DST";
+pub const PVSS_DOM_SEP: &[u8; 21] = b"APTOS_SCRAPE_PVSS_DST"; // TODO: Both names seem poorly chosen
+
+/// Helper trait for deriving random scalars from a transcript.
+///
+/// Not every Fiat–Shamir call needs higher-level operations
+/// (like appending PVSS information), but most do require scalar
+/// derivation. This basic trait provides that functionality.
+///
+/// ⚠️ This trait is intentionally private: `challenge_scalars`
+/// and `challenge_scalar` should **only** be used internally
+/// to ensure consistent scalar generation across protocols.
+trait ScalarProtocol<S: Scalar> {
+    // **Auxiliary** function to return random scalars
+    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<S>;
+    // fn challenge_scalar(&mut self, label: &[u8]) -> S { // Not used yet
+    //     self.challenge_scalars(label, 1)[0]
+    // }
+}
+
+impl<S: Scalar> ScalarProtocol<S> for merlin::Transcript {
+    fn challenge_scalars(&mut self, label: &[u8], num_scalars: usize) -> Vec<S> {
+        let mut buf = vec![0u8; num_scalars * 2 * S::SCALAR_NUM_BYTES];
+        self.challenge_bytes(label, &mut buf);
+
+        let mut result = Vec::with_capacity(num_scalars);
+        for chunk in buf.chunks_exact(2 * S::SCALAR_NUM_BYTES) {
+            debug_assert_eq!(chunk.len(), 2 * S::SCALAR_NUM_BYTES);
+            result.push(S::random_scalar_from_uniform_bytes(chunk));
+        }
+
+        assert_eq!(result.len(), num_scalars);
+        result
+    }
+}
+
+/// Abstraction over prime field scalars used in protocols. Will probably modify or remove this for arkworks
+pub trait Scalar: Copy {
```

Comment:
Can we remove this abstraction please? First, we do not plan to support multiple elliptic curve libraries.
Second, `arkworks` will have its own anyway, should we need an abstraction.